drivers/flash/flash_stm32_qspi.c

```diff
@@ -608,6 +680,133 @@ static int setup_pages_layout(const struct device *dev)
 }
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
+static int qspi_program_addr_4b(const struct device *dev)
+{
+	uint8_t reg;
+	int ret;
+
+	/* Program the flash memory to use 4 bytes addressing */
+	QSPI_CommandTypeDef cmd = {
+		.Instruction = SPI_NOR_CMD_4BA,
+		.InstructionMode = QSPI_INSTRUCTION_1_LINE,
+	};
+
+	ret = qspi_send_cmd(dev, &cmd);
+	if (ret) {
+		return ret;
+	}
+
+	/*
+	 * Read control register to verify if 4byte addressing mode
+	 * is enabled.
+	 */
+	cmd.Instruction = SPI_NOR_CMD_RDCR;
+	cmd.InstructionMode = QSPI_INSTRUCTION_1_LINE;
+	cmd.DataMode = QSPI_DATA_1_LINE;
+
+	ret = qspi_read_access(dev, &cmd, &reg, sizeof(reg));
+	if (ret == 0 && !(reg & SPI_NOR_4BYTE_BIT)) {
+		return -1;
```

Comment:
Shouldn't we adapt some errno ERR here? There is clearly intention for this to have some meaning because it gets printed in log here: https://github.com/zephyrproject-rtos/zephyr/pull/40859/files#diff-39156d185a094793a5d212e3ee4bbc9d275cf3baef6c5eca6aa95089521d1821R861

Reply:
Also in line https://github.com/zephyrproject-rtos/zephyr/pull/40859/files#diff-39156d185a094793a5d212e3ee4bbc9d275cf3baef6c5eca6aa95089521d1821R696 error gets returned from other function, `qspi_send_cmd`, which actually returns negative errno code. So now you have this function either return errno code or hardcoded -1.